packages/myst-transforms/src/html.ts

```diff
@@ -66,6 +98,12 @@ export function htmlTransform(tree: GenericParent, opts?: HtmlTransformOptions)
     node.children = mdast.children as Parent[];
     visit(node, (n: any) => delete n.position);
   });
+  selectAll('paragraph > htmlParsed', tree).forEach((parsed) => {
+    const node = parsed as GenericParent;
+    if (node?.children?.length === 1 && node.children[0].type === 'paragraph') {
+      node.children = node.children[0].children as GenericNode[];
+    }
+  });
```

Comment:
HTML is by default added into a paragraph, if it is already contained in one -- this strips it.